File: pkg/data/models/products/details/cooling.go
package details

import (
	"BetterPC_2.0/pkg/data/models/products/general"
	"go.mongodb.org/mongo-driver/bson/primitive"
)

type Cooling struct {
	ID         primitive.ObjectID `bson:"_id,omitempty"`
	General    general.General    `bson:"general"`
	Type       string             `bson:"type"`
	Sockets    []string           `bson:"sockets"`
	Fans       []int              `bson:"fans"`
	Rpm        []int              `bson:"rpm"`
	Tdp        int                `bson:"tdp"`
	NoiseLevel int                `bson:"noise_level"`
	MountType  string             `bson:"mount_type"`
	Power      int                `bson:"power"`
	Height     int                `bson:"height"`
}

type UpdateCoolingInput struct {
	General    *general.General `bson:"general"`
	Type       *string          `bson:"type"`
	Sockets    *[]string        `bson:"sockets"`
	Fans       *[]int           `bson:"fans"`
	Rpm        *[]int           `bson:"rpm"`
	Tdp        *int             `bson:"tdp"`
	NoiseLevel *int             `bson:"noise_level"`
	MountType  *string          `bson:"mount_type"`
	Power      *int             `bson:"power"`
	Height     *int             `bson:"height"`
}

func (c UpdateCoolingInput) Validate() error {
	return general.ValidateStruct(&c)
}

File: frontend/better_pc/src/modules/products/models/cooling.ts
import {BaseProduct} from "@/modules/products/models/product_common";

export interface Cooling extends BaseProduct {
    type: string; // Тип охлаждения (например, воздушное, жидкостное)
    sockets: string[]; // Совместимые сокеты
    fans: number[]; // Количество вентиляторов
    rpm: number[]; // Скорости вращения вентиляторов
    tdp: number; // Поддерживаемый уровень тепловыделения
    noiseLevel: number; // Уровень шума (в децибелах)
    mountType: string; // Тип крепления
    power: number; // Потребляемая мощность
    height: number; // Высота в мм
}